lance-spark-base_2.12/src/test/java/com/lancedb/lance/spark/read/AggregatePushDownTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lancedb.lance.spark.read;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AggregatePushDownTest {
+  private static SparkSession spark;
+
+  @TempDir static Path tempDir;
+
+  @BeforeAll
+  static void setup() {
+    spark =
+        SparkSession.builder()
+            .appName("LanceAggregatePushDownTest")
+            .master("local[*]")
+            .config("spark.ui.enabled", "false")
+            .config("spark.sql.catalog.lance", "com.lancedb.lance.spark.LanceCatalog")
```

Comment:
all new tests should go with the `LanceNamespaceSparkCatalog`

Reply:
@jackye1995 fixed it

Reply:
LanceNamespaceSparkCatalog is not in lance-spark-base_2.12, and in the packages of various Spark versions, running the unit test will report class not found. 😓